packages/grafana-data/src/dataframe/StreamingDataFrame.ts

```diff
@@ -3,12 +3,70 @@ import { QueryResultMeta } from '../types';
 import { ArrayVector } from '../vector';
 import { DataFrameJSON, decodeFieldValueEntities } from './DataFrameJSON';
 
+// binary search for index of closest value
+function closestIdx(num: number, arr: number[], lo?: number, hi?: number) {
+  let mid;
+  lo = lo || 0;
+  hi = hi || arr.length - 1;
+  let bitwise = hi <= 2147483647;
+
+  while (hi - lo > 1) {
+    mid = bitwise ? (lo + hi) >> 1 : Math.floor((lo + hi) / 2);
+
+    if (arr[mid] < num) {
+      lo = mid;
+    } else {
+      hi = mid;
+    }
+  }
+
+  if (num - arr[lo] <= arr[hi] - num) {
+    return lo;
+  }
+
+  return hi;
+}
+
+// mutable circular push
+function circPush(data: number[][], newData: number[][], maxLength = Infinity, deltaIdx = 0, maxDelta = Infinity) {
+  for (let i = 0; i < data.length; i++) {
+    data[i] = data[i].concat(newData[i]);
+  }
+
+  const nlen = data[0].length;
```

Comment:
switched this 0 since always accurate and `deltaIdx` will be -1 if there is no time field!